### ISSUE TITLE
Error when using polymorphic relationships

### ISSUE THREAD

**Issue by Kislax** [prisma/prisma#19547]
### Bug description

I created a schema with polymorphic links.
My goal is to have foreign keys of different services connected to one field, and define the service by the Servicetype field.

But when I try to create an entry in c bound id from table gitlabService. Prisma always tries to find the id in the most recent foreign key (it's confluence right now) and get an error

### How to reproduce

```bash
prisma:info Starting a postgresql pool with 13 connections.
prisma:query BEGIN
prisma:query INSERT INTO "prfactory"."Products" ("name","key","is_archived") VALUES ($1,$2,$3) RETURNING "prfactory"."Products"."id"
prisma:query SELECT "prfactory"."Products"."id", "prfactory"."Products"."name", "prfactory"."Products"."key", "prfactory"."Products"."is_archived" FROM "prfactory"."Products" WHERE "prfactory"."Products"."id" = $1 LIMIT $2 OFFSET $3
prisma:query COMMIT
prisma:query BEGIN
prisma:query INSERT INTO "prfactory"."GitLabServices" ("group_id","group_url","group_name","group_path","group_json") VALUES ($1,$2,$3,$4,$5) RETURNING "prfactory"."GitLabServices"."id"
prisma:query SELECT "prfactory"."GitLabServices"."id", "prfactory"."GitLabServices"."group_id", "prfactory"."GitLabServices"."group_url", "prfactory"."GitLabServices"."group_name", "prfactory"."GitLabServices"."group_path", "prfactory"."GitLabServices"."group_json" FROM "prfactory"."GitLabServices" WHERE "prfactory"."GitLabServices"."id" = $1 LIMIT $2 OFFSET $3
prisma:query COMMIT
{
  id: 10n,
  groupId: 77777n,
  groupUrl: '/f
                                                                                                                        ',
  groupName: 'test
                                                                                                                         ',
  groupPath: 'test
                                                                                                                         ',
  groupJson: '123'
}
{
  id: 10n,
  name: 'prod1
                                                                                                                    ',
  key: 'key1
                                                                                                                   ',
  isArchived: false
}
prisma:query BEGIN
prisma:query INSERT INTO "prfactory"."ProductToService" ("serviceType","product_id","service_id") VALUES ($1,$2,$3) RETURNING "prfactory"."ProductToService"."id"
prisma:query ROLLBACK
prisma:error 
Invalid `prisma.productToService.create()` invocation in
C:\Users\mapodstrec\Develop\PF-BE\src\models\gitlab\model.ts:416:48

  413 console.log(git)
  414 console.log(product)
  415
→ 416 const item = await prisma.productToService.create(
Foreign key constraint failed on the field: `ProductToService_confluenceSpaces_id_fk (index)`
```

### Expected behavior

an entry will be created in the productToService table with the associated GitlabService

### Prisma information

```prisma
// Add your schema.prisma
```
model Products {
  id         BigInt  @id @unique @default(autoincrement()) // Идентификатор записи. Первичный ключ
  name       String  @db.Char(255) // Имя проекта
  key        String  @db.Char(255) // Ключ проекта
  isArchived Boolean @default(false) @map("is_archived") @db.Boolean // Флаг того, что продукт архивирован. Удаления продуктов не предусмотрено.

  productToService ProductToService[]
  productToUser    ProductToUser[]
  serviceInstance  ServiceInstances?
}

model JiraProjects {
  id     BigInt @id @unique @default(autoincrement()) // Идентификатор записи. Первичный ключ
  name   String @db.Char(255) // Имя проекта
  key    String @db.Char(255) // Ключ проекта
  url    String @db.Char(255) // self-url на проект в Jira
  json   Json   @db.Json // json полной информации по проекту. Запоминается при создании
  jiraId BigInt @map("jira_id") // Идентификатор проекта в JIRA

  productToService ProductToService[] @relation("jiraService")
}

model GitLabServices {
  id        BigInt @id @unique @default(autoincrement()) // Идентификатор записи. Первичный ключ
  groupId   BigInt @map("group_id") // Идентификатор группы в GitLab
  groupUrl  String @map("group_url") @db.Char(255) //Ссылка на группу в GitLab
  groupName String @map("group_name") @db.Char(255) //Наименование группы в GitLab
  groupPath String @map("group_path") @db.Char(255) //PATH группы в GitLab
  groupJson Json   @map("group_json") @db.Json //json группы из GitLab

  productToService ProductToService[] @relation("gitlabService")
}

model HarborServices {
  id        BigInt @id @unique @default(autoincrement()) // Идентификатор записи. Первичный ключ
  projectId BigInt @map("project_id") // Идентификатор проекта в Harbor
  name      String @db.Char(255) //Имя проекта
  url       String @db.Char(255) //self-url на проект в Harbor
  json      Json   @db.Json //json полной информации по проекту.

  productToService ProductToService[] @relation("harborService")
}

model SonarQubeServices {
  id    BigInt @id @unique @default(autoincrement()) // Идентификатор записи. Первичный ключ
  token String @db.Char(255) //Токен в SQ
  url   String @db.Char(255) //self-url на проект в SQ

  productToService ProductToService[] @relation("sonarqubeService")
}

model ConfluenceSpaces {
  id       BigInt @id @unique @default(autoincrement()) // Идентификатор записи. Первичный ключ
  spaceId  BigInt @map("space_id") // 	Идентификатор пространства в Confluence
  spaceUrl String @map("space_url") @db.Char(255) //Url созданного пространства в Confluence
  spaceKey String @map("space_key") @db.Char(255) //Ключ пространства в Confluence
  homepage String @db.Char(255) //Идентификатор домашней страницы созданного пространства
  json     Json   @db.Json //json полной информации по пространству.

  productToService ProductToService[] @relation("confluenceService")
}

enum ServiceType {
  GITLAB
  JIRA
  HARBOR
  CONF
  SONAR
}

model ProductToService {
  id          BigInt      @id @unique @default(autoincrement()) // Идентификатор записи. Первичный ключ
  serviceType ServiceType // Тип сервиса
  productId   BigInt      @map("product_id") // Идентификатор продукта
  serviceId   BigInt      @map("service_id") // Идентификатор продукта в jira

  products          Products           @relation(fields: [productId], references: [id])
  gitLabServices    GitLabServices?    @relation(name: "gitlabService", fields: [serviceId], references: [id], map: "ProductToService_gitLabServices_id_fk")
  jiraProjects      JiraProjects?      @relation(name: "jiraService", fields: [serviceId], references: [id], map: "ProductToService_jiraServices_id_fk")
  harborServices    HarborServices?    @relation(name: "harborService", fields: [serviceId], references: [id], map: "ProductToService_harborServices_id_fk")
  sonarQubeServices SonarQubeServices? @relation(name: "sonarqubeService", fields: [serviceId], references: [id], map: "ProductToService_sonarQubeServices_id_fk")
  confluenceSpaces  ConfluenceSpaces?  @relation(name: "confluenceService", fields: [serviceId], references: [id], map: "ProductToService_confluenceSpaces_id_fk")

  @@index([serviceType, serviceId], name: "service_type_and_id")
```ts
// Add your code using Prisma Client
static async blo () {
    const product = await prisma.products.create({
      data: {
        name: 'prod1',
        key: 'key1',
        isArchived: false
      }
    })

    const git = await prisma.gitLabServices.create({
      data: {
        groupId: 77777,
        groupName: 'test',
        groupPath: 'test',
        groupUrl: '/f',
        groupJson: '123'
      }
    })
    console.log(git)
    console.log(product)

    const item = await prisma.productToService.create({
      data: {
        serviceType: 'GITLAB',
        productId: BigInt(1),
        serviceId: BigInt(git.id)
      }
    })
    return item
  }
```

OR

```typescript
const item = await prisma.productToService.create({
        data: {
          serviceType: 'GITLAB',
          products: {
            create: {
              name: 'prod1',
              key: 'key1',
              isArchived: false
            }
          },
          gitLabServices: {
            create: {
              groupId: 77777,
              groupName: 'test',
              groupPath: 'test',
              groupUrl: '/f',
              groupJson: '123'
            }
          }
        }
      })
```

### Environment & setup

- OS:  Windows
- Database: PostgreSQL
- Node.js version: v16.18.1

### Prisma Version

```
prisma                  : 4.14.0
@prisma/client          : 4.14.0
Current platform        : windows
Query Engine (Node-API) : libquery-engine d9a4c5988f480fa576d43970d5a23641aa77bc9c (at node_modules\prisma\node_modules\@prisma\engines\query_engine-windows.dll.node)      
Migration Engine        : migration-engine-cli d9a4c5988f480fa576d43970d5a23641aa77bc9c (at node_modules\prisma\node_modules\@prisma\engines\migration-engine-windows.exe)  
Format Wasm             : @prisma/prisma-fmt-wasm 4.14.0-67.d9a4c5988f480fa576d43970d5a23641aa77bc9c
Default Engines Hash    : d9a4c5988f480fa576d43970d5a23641aa77bc9c
Studio                  : 0.484.0
```

**Comment by FebriTahta** [comment]
this is solved ?

**Comment by Kislax** [comment]
No, at first they used individual fields for tables, and then they made a separate table of connections and identifiers and when requesting, they simply made a separate call from this table. Now I can’t say the status, because I have not been working on the project for a long time
